clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryExporterProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.telemetry;
+
+import org.apache.kafka.common.metrics.MetricsReporter;
+
+/**
+ * A {@link MetricsReporter} may implement this interface to indicate support for collecting client
+ * telemetry on the server side using the new exporter API. This interface replaces the deprecated
```

Comment:
I think it would be better not to include mention of the deprecated `ClientTelemetry` in the comment of the new interface. In AK 5.0, we will chop out the old interface and might well leave comments such as these by accident.

Reply:
Can we also remove the mention of the old API in `ClientTelemetryExporter`?
Also let's update the javadoc of `ClientTelemetryPayload` to mention `ClientTelemetryExporter`